GitHubRepositories.ps1

```diff
@@ -2850,3 +2853,189 @@ filter Add-GitHubRepositoryAdditionalProperties
+        {
+            $UserName = $item.login
+            if ([String]::IsNullOrEmpty($UserName) -and $PSBoundParameters.ContainsKey('Name'))
+            {
+                $UserName = $Name
+            }
+
+            if (-not [String]::IsNullOrEmpty($UserName))
+            {
+                $addMemberParms = @{
+                    InputObject = $item
+                    Name = 'UserName'
+                    Value = $UserName
+                    MemberType = 'NoteProperty'
+                    Force = $true
+                }
+                Add-Member @addMemberParms
+            }
+
+            $UserId = $item.id
+            if (($UserId -eq 0) -and $PSBoundParameters.ContainsKey('Id'))
+            {
+                $UserId = $Id
+            }
+
+            if ($UserId -ne 0)
+            {
+                $addMemberParms = @{
+                    InputObject = $item
+                    Name = 'UserId'
+                    Value = $UserId
+                    MemberType = 'NoteProperty'
+                    Force = $true
+                }
+
+                Add-Member @addMemberParms
+            }
+        }
+
+        Write-Output $item
+    }
+}
+
+filter Add-GitHubRepositoryCollaboratorAdditionalProperties
+{
+<#
+    .SYNOPSIS
+        Adds type name and additional properties to ease pipelining to GitHub Collaborator objects.
+
+    .PARAMETER InputObject
+        The GitHub object to add additional properties to.
+
+    .PARAMETER TypeName
+        The type that should be assigned to the object.
+
+    .PARAMETER Name
+        The name of the Collaborator.  This information might be obtainable from InputObject, so this
+        is optional based on what InputObject contains.
+
+    .PARAMETER Id
+        The ID of the Collaborator.  This information might be obtainable from InputObject, so this
+        is optional based on what InputObject contains.
+
+    .INPUTS
+        PSCustomObject
+
+    .OUTPUTS
+        GitHub.RepositoryCollaborator
+#>
+    [CmdletBinding()]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseSingularNouns', '',
+        Justification='Internal helper that is definitely adding more than one property.')]
+    param(
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline)]
+        [AllowNull()]
+        [AllowEmptyCollection()]
+        [PSCustomObject[]] $InputObject,
+
+        [ValidateNotNullOrEmpty()]
+        [string] $TypeName = $script:GitHubRepositoryCollaboratorTypeName,
+
+        [string] $Name,
+
+        [int64] $Id
+    )
+
+    foreach ($item in $InputObject)
+    {
+        $item.PSObject.TypeNames.Insert(0, $TypeName)
+
+        if (-not (Get-GitHubConfiguration -Name DisablePipelineSupport))
+        {
+            $UserName = $item.login
+            if ([String]::IsNullOrEmpty($UserName) -and $PSBoundParameters.ContainsKey('Name'))
+            {
+                $UserName = $Name
```

Comment:
nit: camelCase since it's a local var.  Here and with its reference below.  Same for `UserId`.  Same for other function above.

```suggestion
                $userName = $Name
```

Reply:
Done